packages/astro-rss/src/index.ts

```diff
@@ -210,15 +206,19 @@ async function generateRSS(rssOptions: ValidatedRSSOptions): Promise<string> {
 		);
 	// items
 	root.rss.channel.item = items.map((result) => {
-		// If the item's link is already a valid URL, don't mess with it.
-		const itemLink = isValidURL(result.link)
-			? result.link
-			: createCanonicalURL(result.link, rssOptions.trailingSlash, site).href;
-		const item: any = {
-			title: result.title,
-			link: itemLink,
-			guid: { '#text': itemLink, '@_isPermaLink': 'true' },
-		};
+		const item: Record<string, unknown> = {};
+
+		if (result.title) {
+			item.title = result.title;
+		}
+		if (result.link) {
+			// If the item's link is already a valid URL, don't mess with it.
+			const itemLink = isValidURL(result.link)
+				? result.link
+				: createCanonicalURL(result.link, rssOptions.trailingSlash, site).href;
+			item.link = itemLink;
+			item.guid = { '#text': itemLink, '@_isPermaLink': 'true' };
```

Comment:
Since it depends on link, I add it here. But I don't know if this property is supposed to be something else when there is no link